crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -456,11 +456,25 @@ impl Instruction {
             Operation::Not(value) => {
                 if let Some(l_const) = eval_fn(ctx, *value).into_const_value() {
                     let l = self.res_type.field_to_type(l_const).to_u128();
-                    if self.res_type.bits() != 32 {
-                        unimplemented!();
+                    let bit_string = format!("{:b}", l);
```

Comment:
Converting to a String and doing the bitwise not on each character seems rather inefficient, what is preventing us from doing `!l % max_value`?

Reply:
Indeed, good point.